src/Scripts/Game/Components/Vehicle/EL_SirenManagerComponent.c

```diff
@@ -0,0 +1,370 @@
+	/**
+	\brief Sets the signals of the knob procedural animation. Called from SetMode()
+	\param mode - the mode that is going to be set
+	**/
+	protected void SetKnobMode(EL_SirenMode mode)
+	{
+		EL_LightAnimation anim = mode.GetAnimation();
+		if(anim)
+		{
+			anim.Reset();
+			m_Knob.SetAnimation(anim);
+		}
+		foreach(SignalValuePair sig : mode.GetKnobSignals())
+		{
+			m_KnobSigComp.SetSignalValue(m_KnobSigComp.FindSignal(sig.GetName()), sig.GetValue());
+		}
+		m_SoundComp.SoundEventBone("SOUND_VEHICLE_CLOSE_LIGHT_ON", "Scene_Root");
+		
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	/**
+	\brief Sets the current mode based on m_CurrentModeIndex. Changes knob position and siren sounds
+	**/
+	protected void SetMode()
+	{
+		EL_SirenMode mode = m_Modes.GetMode(m_CurrentModeIndex);
+		m_SoundComp.SoundEvent("Vehicle_Horn");
+		if(mode)
+		{
+			if(m_Knob) SetKnobMode(mode);
+			
+			if(m_SoundComp) SetSirenMode(mode);
+			if(mode.GetName() != "default") 
+			{
+				m_SoundComp.SetSignalValueStr("animSelector", 1);
+				m_SoundComp.SoundEventBone("SOUND_VEHICLE_HORN", "Scene_Root");
+				
+				m_SoundComp.SetSignalValueStr("animSelector", 0);
+				
+			}
+		}
+		
+		m_CurrentMode = mode;
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	/**
+	\brief Sets mode by name. See SetMode()
+	\param name - name of the mode
+	**/
+	void SetModeStr(string name)
+	{
+		if(!m_Modes) return;
+		int nextModeIndex = m_Modes.Find(name);
+		if(nextModeIndex == m_CurrentModeIndex) return;
+		m_CurrentModeIndex = nextModeIndex;
+		SetMode();
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	override bool RplSave(ScriptBitWriter writer)
+	{
+		writer.WriteInt(m_CurrentModeIndex);
+		writer.WriteFloat(m_CurrentMode.GetAnimation().GetTimeInLoop());
+		return true;
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	override bool RplLoad(ScriptBitReader reader)
+	{
+		float timeSlice;
+		reader.ReadInt(m_CurrentModeIndex);
+		SetMode();
+		reader.ReadFloat(timeSlice);
+		m_CurrentMode.GetAnimation().Tick(timeSlice);
+		return true;
+	}
+}
+
+
+/**
+Contains all the information about the siren mode. Animation, sounds and knob procedural animation signals
+**/
+[BaseContainerProps(), SCR_BaseContainerCustomTitleField("m_Name", "%1")]
+class EL_SirenMode
+{
+	// Available modes to the user
+	static const ref ParamEnumArray MODE_NAMES =
+	{
+		new ParamEnum("default", "0"),
+		new ParamEnum("Mode1", "1"),
+		new ParamEnum("Mode2", "2"),
+		new ParamEnum("Mode3", "3"),
+		new ParamEnum("Mode4", "4"),
+		new ParamEnum("Mode5", "5"),
+		new ParamEnum("Mode6", "6"),
+		new ParamEnum("Mode7", "7")
+	};
+	
+	// Available sounds
+	static protected const ref ParamEnumArray SIREN_SOUNDS = 
```

Comment:
Previous array is not protected because `EL_SirenModeAction` needs to access the available modes, while the sounds are only selected in `EL_SirenMode` class